### PR TITLE
User created Tag names are now only allows in lowercase. #80

### PR DESCRIPTION
**Pull request by maguero**
This PR fixes issue #80.

### CODE REVIEW

File: util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.killbill.billing.util.tag.dao.TagDefinitionDao;
+import org.killbill.billing.util.tag.dao.TagDefinitionModelDao;
+import org.mockito.Mockito;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Created by maguero on 14/10/15.
```
Comment (pierre) [inline util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java]: Could you remove this?

File: util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+public class DefaultTagUserApiTest extends UtilTestSuiteNoDB {
+
+    private DefaultTagUserApi tagUserApi;
+    private TagDefinitionDao tagDefinitionDao;
+    private CallContext context;
+
+    @BeforeMethod
```
Comment (pierre) [inline util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java]: All TestNG annotations need to specify a test group, either `fast` (when extending `UtilTestSuiteNoDB`, like here) or `slow` (when extending `UtilTestSuiteWithEmbeddedDB`).

File: util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        final TagDao tagDao = Mockito.mock(TagDao.class);
+        final InternalCallContextFactory internalCallContextFactory = Mockito.mock(InternalCallContextFactory.class);
+        tagDefinitionDao = Mockito.mock(TagDefinitionDao.class);
+        tagUserApi = new DefaultTagUserApi(internalCallContextFactory, tagDefinitionDao, tagDao);
+        context = Mockito.mock(DefaultCallContext.class);
+    }
+
+    @Test(expectedExceptions = TagDefinitionApiException.class, expectedExceptionsMessageRegExp = "The tag definition name must be in lowercase .*")
+    public void testCreateTagDefinitionWithUpperCase() throws Exception {
+        tagUserApi.createTagDefinition("someUpperCaseLetters", "description", context);
```
Comment (pierre) [inline util/src/test/java/org/killbill/billing/util/tag/api/DefaultTagUserApiTest.java]: Could you add 2 other tests with one letter at the front and one at the back, like `Invalid` and `invaliD`? This is just to validate the regex.

**Comment by maguero** [comment]
Test class udpated.

**Comment by pierre** [comment]
:tada: Awesome, thanks!

**Comment by pierre** [comment]
I've pushed https://github.com/killbill/killbill/commit/ad83186b97cb1b63c2716a4b0bde988858c0275f to fix an existing test.

FYI, you can run all regression tests by running: `mvn clean install -Pmysql`.

**Comment by maguero** [comment]
great, goot to know.
thank you.

**Pull request merged**